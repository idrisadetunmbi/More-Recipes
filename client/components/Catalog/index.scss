#catalog-component {
  #main-divider {
    margin-left: 3em;
    margin-right: 3em; 
    margin-top: 0; 
    display: block;
  }
  
  #search-section {
    margin: 3em;
    margin-bottom: 0;
    input {
      margin-left: -1em;
      border: 1px solid #9e9e9e;
      padding-left: .8em;
      font-size: 1em;
      font-family: 'Lato';
      border-radius: .25rem;
    };
    div {
      position: relative;
    }
    i {
      position: absolute;
      top: 0.7rem;
      right: 1.5rem;
      cursor: pointer;
      display: none;
    }
  };

  #suggestion-section {
    padding: 3em;
    margin-top: 1em;
    .divider {
      margin-bottom: 2em;
    }
  };

  h5 {
    font-family: 'Raleway';
    font-style: italic;
    font-size: 1.5rem;
    padding-left: .5rem;
    text-align: center;
  };

  #gallery-before {
    margin: 0 .7rem 1rem .5rem;
  }

  .infinite-scroll {
    position: absolute;
    bottom: -3rem;
    margin: 0 auto;
    text-align: center;
    left: 0;
    right: 0;    
  }
}