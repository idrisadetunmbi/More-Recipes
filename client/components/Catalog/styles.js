export const searchInput = {
  marginLeft: '-1em',
  border: '1px solid #9e9e9e',
  height: '2.5em',
  width: '65%',
  marginRight: '0',
  paddingLeft: '.8em',
  fontSize: '1.5em',
  fontFamily: 'Open Sans',
  fontStyle: 'italic',
};

export const searchInputEnclosingDiv = {
  margin: '3em',
  marginBottom: '0',
};
