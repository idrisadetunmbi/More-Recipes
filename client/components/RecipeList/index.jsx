import React from 'react';
import { Link } from 'react-router-dom';

import Recipe from '../Recipe';

export const RecipeList = (props) => {
  const { recipes, isLoadingRecipes, error } = props;
  const { style, gridStyle } = props;
  return (
    isLoadingRecipes ?
      <div>Loading Recipes...</div> :
      <div className="row" id="recipe-gallery" style={style}>
        {
          // eslint-disable-next-line
          error ?
            <div className="center">There was an error loading recipes - {error.error}</div> :
            recipes.length ?
              recipes.map(recipe =>
               (
                 <Link key={recipe.id} to={`/recipes/${recipe.id}`}>
                   <Recipe gridStyle={gridStyle} recipe={recipe} />
                 </Link>)) :
               (<p>There are no recipes in this category</p>)
        }
      </div>
  );
};


export default RecipeList;
