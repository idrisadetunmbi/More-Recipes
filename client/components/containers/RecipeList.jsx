import { connect } from 'react-redux';
import RecipeListComponent from '../RecipeList';
import { fetchRecipes } from '../../actions/recipes';

const mapStateToProps = (state) => {
  return {
    recipes: state.recipes,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchRecipes: () => dispatch(fetchRecipes()),
  };
};

const RecipeList = connect(mapStateToProps, mapDispatchToProps)(RecipeListComponent);

export default RecipeList;
