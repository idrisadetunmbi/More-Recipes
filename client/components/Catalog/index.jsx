import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import RecipeList from '../RecipeList';
import './index.scss';


/**
 * @param {any} props
 *
 * @returns {Object} Catalog DOM Node
 */
export const Catalog = props => (
  <div id="catalog-component">
    <div id="search-section" className="row">
      <div className="col l3 m4 s12">
        <button className="btn-large">Categories</button>
      </div>
      <div className="col l8 m8 s12">
        <input placeholder="Search Recipe" type="text" />
        <a href="#" className="btn-large">
          <i className="material-icons medium">search</i>
        </a>
      </div>
    </div>

    <div id="main-divider" className="divider" />

    <div className="row">
      <div className="col l3" id="suggestion-section">
        <h5>Today{'\''}s Combo</h5>
        <div className="divider" />
        <div className="recipe">
          <div className="card">
            <div className="card-image">
              <img alt="" src="images/sample_recipes/recipe0116-xl-farro-with-vinegar-glazed-sweet-potato-and-apples.jpg" />
            </div>
            <div className="card-content">
              <span className="card-title">Farrow with vinegar, glazed sweet potato and apples</span>
              <p>I am a very simple card. I am good at containing small&nbsp;
                bits of information. I am convenient because I require&nbsp;
                little markup to use effectively.
              </p>
            </div>
            <div className="card-action">
              <a href="#">This is a link</a>
            </div>
          </div>
        </div>
      </div>

      <div className="col l9 s12">
        <h5>Featured Recipes</h5>
        <div className="divider" id="gallery-before" />
        <RecipeList
          recipes={props.recipes.recipes}
          isLoadingRecipes={props.recipes.requestInitiated}
          error={props.recipes.requestError}
          style={{ width: '97%', marginLeft: '-0.6em' }}
          gridStyle="l4 s12 m6"
        />
      </div>
    </div>

    <div className="fixed-action-btn">
      <a
        className="btn-floating btn-large"
        onClick={() => {
        const { history } = props;
        if (!props.user.data.token) {
          history.push('/signin', {
            modal: true,
            previousLocation: history.location.pathname,
          });
          return;
        }
        history.push('/recipes/create', {
          modal: true,
          previousLocation: history.location.pathname,
        });
      }}
      >
        <i className="material-icons">add</i>
      </a>
    </div>
  </div>
);

const mapStateToProps = state =>
  ({
    user: state.user,
    recipes: state.recipes,
  });

Catalog.propTypes = {
  recipes: PropTypes.shape({
    recipes: PropTypes.array,
    requestInitiated: PropTypes.bool,
    requestError: PropTypes.object,
  }).isRequired,

  history: PropTypes.shape({
    location: PropTypes.shape({
      pathname: PropTypes.string,
    }),
  }).isRequired,

  user: PropTypes.shape({
    data: PropTypes.shape({
      token: PropTypes.string,
    }),
  }).isRequired,
};

export default connect(mapStateToProps)(Catalog);
