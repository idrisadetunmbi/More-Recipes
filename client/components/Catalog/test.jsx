import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Catalog } from '../Catalog';

Enzyme.configure({ adapter: new Adapter() });

describe('<Catalog />', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<Catalog recipes={{ recipes: [] }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('simulates click events', () => {
    const wrapper = mount(<Catalog recipes={{ recipes: [] }} user={{ data: { } }} history={{ location: { pathname: '' }, push: jest.fn() }} />);
    const addRecipeOnClick = jest.spyOn(wrapper.instance(), 'addRecipeOnClick');
    wrapper.instance().forceUpdate();
    wrapper.find('.btn-floating').simulate('click');
    expect(addRecipeOnClick).toHaveBeenCalled();
  });
});
