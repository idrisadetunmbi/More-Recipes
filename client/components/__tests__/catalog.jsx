import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Catalog } from '../Catalog';
import { RecipeList } from '../RecipeList';

Enzyme.configure({ adapter: new Adapter() });

describe('<Catalog />', () => {
  it('matches snapshot', () => {
    const component = renderer.create(<Catalog recipes={{ recipes: [] }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders <RecipeList /> component', () => {
    const wrapper = shallow(<Catalog recipes={{ recipes: [] }} />);
    expect(wrapper.find(RecipeList)).toBeTruthy();
  });

  it('simulates click events', () => {
    const addRecipeButton = jest.fn();
    const wrapper = shallow(<Catalog recipes={{ recipes: [] }} />);
    try { wrapper.find('.btn-floating').simulate('click'); } catch (error) {}
    expect(addRecipeButton).toHaveBeenCalled();
  });
});
