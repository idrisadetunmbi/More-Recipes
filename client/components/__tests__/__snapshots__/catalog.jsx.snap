// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Catalog /> matches snapshot 1`] = `
<div
  id="catalog-component"
>
  <div
    className="row"
    id="search-section"
  >
    <div
      className="col l3 m4 s12"
    >
      <button
        className="btn-large"
      >
        Categories
      </button>
    </div>
    <div
      className="col l8 m8 s12"
    >
      <input
        placeholder="Search Recipe"
        type="text"
      />
      <a
        className="btn-large"
        href="#"
      >
        <i
          className="material-icons medium"
        >
          search
        </i>
      </a>
    </div>
  </div>
  <div
    className="divider"
    id="main-divider"
  />
  <div
    className="row"
  >
    <div
      className="col l3"
      id="suggestion-section"
    >
      <h5>
        Today
        '
        s Combo
      </h5>
      <div
        className="divider"
      />
      <div
        className="recipe"
      >
        <div
          className="card"
        >
          <div
            className="card-image"
          >
            <img
              alt=""
              src="images/sample_recipes/recipe0116-xl-farro-with-vinegar-glazed-sweet-potato-and-apples.jpg"
            />
          </div>
          <div
            className="card-content"
          >
            <span
              className="card-title"
            >
              Farrow with vinegar, glazed sweet potato and apples
            </span>
            <p>
              I am a very simple card. I am good at containing small  bits of information. I am convenient because I require  little markup to use effectively.
            </p>
          </div>
          <div
            className="card-action"
          >
            <a
              href="#"
            >
              This is a link
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col l9 s12"
    >
      <h5>
        Featured Recipes
      </h5>
      <div
        className="divider"
        id="gallery-before"
      />
      <div
        className="row"
        id="recipe-gallery"
        style={
          Object {
            "marginLeft": "-0.6em",
            "width": "97%",
          }
        }
      >
        <p>
          There are no recipes in this category
        </p>
      </div>
    </div>
  </div>
  <div
    className="fixed-action-btn"
  >
    <a
      className="btn-floating btn-large"
      onClick={[Function]}
    >
      <i
        className="material-icons"
      >
        add
      </i>
    </a>
  </div>
</div>
`;
